### PR TITLE
nixos/syncoid: fix permissions without --no-sync-snap

### PR DESCRIPTION
**Pull request by danielfullmer**
After 733acfa140d5b73bc69c53c4ebd90ccc5f281f0e, syncoid would fail to
run if `commonArgs` did not include `[ "--no-sync-snap" ]`, since it would
not have permissions to create or destroy snapshots.

CC @lopsided98 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/backup/syncoid.nix [hunk 1]
```diff
@@ -197,14 +197,14 @@ in {
...
-            "+/run/booted-system/sw/bin/zfs" "allow"
-            cfg.user "create,mount,receive,rollback" pool
-          ]) (getPools "target"));
+          ExecStartPre = let
+            allowCmd = permissions: pool: lib.escapeShellArgs [
+              "+/run/booted-system/sw/bin/zfs" "allow"
+              cfg.user (concatStringsSep "," permissions) pool
+            ];
+          in
+            (map (allowCmd ([ "hold" "send" ] ++ optionals (!(elem "--no-sync-snap" cfg.commonArgs)) [ "snapshot" "destroy" ])) (getPools "source")) ++
```
Comment (lopsided98) [inline nixos/modules/services/backup/syncoid.nix:206]: If we do this conditionally, the logic needs to be more complicated and incorporate the `extraArgs` and `useCommonArgs` options. On the other hand, perhaps we should just add these permissions unconditionally.

**Review by lopsided98 — CHANGES_REQUESTED** [review]
Can you guess what argument every command in my config has... :)

File: nixos/modules/services/backup/syncoid.nix (hunk 1 above)
Comment (danielfullmer) [reply nixos/modules/services/backup/syncoid.nix]: Pushed a new change where I opted to just add those permissions unconditionally, for simplicity.

**Review by lopsided98 — CHANGES_REQUESTED** [review]
Thinking about this some more, I would rather test both with and without `--no-sync-snap`. I have implemented it here: https://github.com/lopsided98/nixpkgs/commit/7052ab7dbf34d5a81eb5caab0d9d80a7ba665fc6, if you want to pull it into this PR.

**Review by lopsided98 — APPROVED** [review]
cc @bhipple, since you reviewed #79759, which this PR fixes.

**Comment by danielfullmer** [comment]
@lopsided98 Also fixed a small typo in your commit message: s/synoid/syncoid/

**Comment by lopsided98** [comment]
@ofborg test sanoid

**Pull request merged**